src/Nethermind/Nethermind.Evm/StateOverridesExtensions.cs

```diff
@@ -43,9 +41,12 @@ public static void ApplyStateOverrides(
             }
         }
 
-        state.Commit(spec);
-        state.CommitTree(blockNumber);
-        state.RecalculateStateRoot();
+        if (commit)
```

Comment:
`eth_simulate` implementation does commit after an override, but looks like other methods do not need it. Still testing all the cases and will resolve this once sure.